src/Extractors/Extractor.php

```diff
@@ -20,14 +26,61 @@ public static function fromFile($file, Translations $translations = null)
         if ($translations === null) {
             $translations = new Translations();
         }
+        $originalSIC = $translations->getSkipIntegrityChecks();
+        if (static::$skipIntegrityChecksOnImport) {
```

Comment:
Use `static` instead of `self` to use the class-specific values (see http://3v4l.org/rWAq8 )